scripts/promote-catalog.py

```diff
@@ -0,0 +1,493 @@
+
+from operatorcert import github, ocp_version_info
+from operator_repo import Repo as OperatorRepo
+
+LOG = logging.getLogger(__name__)
+
+ORGANIZATIONS = {
+    "community-test-repo": {
+        "gh_repository": "mantomas/community-test-repo",
+        "organization": "community-operators",
+    },
+    "certified-operators": {
+        "gh_repository": "redhat-openshift-ecosystem/certified-operators",
+        "organization": "certified-operators",
+    },
+    "redhat-marketplace-operators": {
+        "gh_repository": "redhat-openshift-ecosystem/redhat-marketplace-operators",
+        "organization": "redhat-marketplace",
+    },
+    "community-operators-prod": {
+        "gh_repository": "redhat-openshift-ecosystem/community-operators-prod",
+        "organization": "community-operators",
+    },
+    "community-operators-pipeline-preprod": {
+        "gh_repository": "redhat-openshift-ecosystem/community-operators-pipeline-preprod",
+        "organization": "community-operators",
+    },
+}
+
+
+def setup_argparser() -> Any:
+    """
+    Setup a argument parser for the script
+
+    Returns:
+        Any: Argument parser
+    """
+    parser = argparse.ArgumentParser(
+        description="Promote catalog to the target version, defaults to latest supported. "
+        "You need to have GITHUB_TOKEN exported in the environment to create PRs. "
+        "Author of the PRs will be the owner of the token."
+    )
+    parser.add_argument(
+        "--local-repo", help="Local repository to be processed (abs path)"
+    )
+    parser.add_argument("--target-version", help="Version to promote to (eg. 4.16)")
+    parser.add_argument(
+        "--pyxis-url",
+        default="https://catalog.redhat.com/api/containers/",
+        help="Base URL for Pyxis container metadata API",
+    )
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Dry run mode - branches will be pushed without creating PRs. "
+        "Can be used to avoid multiple pipelines starting at once",
+        default=False,
+    )
+
+    return parser
+
+
+def copy_files_if_not_exist(
+    operator_name: str,
+    repo_dir: Path,
+    source_version: str,
+    target_version: str,
+) -> list[str]:
+    """
+    Copy catalog and template files if they don't exist
+
+    Args:
+        operator_name (str): Operator name
+        repo_dir (Path): Path to the processed local repository
+        source (str): Source catalog directory
+        target (str): Target catalog directory
+
+    Returns:
+        list[str]: List of relative paths to copied files
+        for Git commit
+    """
+    to_commit = []
+
+    # copy catalog files if they don't exist
+    catalog_dir = repo_dir / "catalogs"
+    catalog_target_dir = catalog_dir / f"v{target_version}" / operator_name
+    catalog_target_dir.mkdir(mode=755, parents=True, exist_ok=True)
```

Comment:
755 != 0o755:

```
$ python
Python 3.12.3 (main, Apr 17 2024, 00:00:00) [GCC 13.2.1 20240316 (Red Hat 13.2.1-7)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 0o755
493
>>> 
```

Unix permissions are expressed as octal numbers. 755 octal means rwx-rx-rx permissions, 755 decimal is just a mess :)

Reply:
Ah, sorry I missed that - thanks for clarification! Reverted to 0o755.